constants.js

```diff
@@ -10,6 +10,9 @@ module.exports = {
   VERTICAL_ORIENTATION: 'vertical',
   VERTICAL_SCROLLABLE: 'verticalScrollable',
 
+  BEFORE_POSITION: 'before',
```

Comment:
Can we make this even more specific and call it `ICON_BEFORE_POSITION` or something. even `ICON_BEFORE` would be fine.